lib/service/kubernetes.go

```diff
@@ -284,9 +284,7 @@ func (process *TeleportProcess) initKubernetesService(log *logrus.Entry, conn *C
 
 	// Cleanup, when process is exiting.
 	process.OnExit("kube.shutdown", func(payload interface{}) {
-		if asyncEmitter != nil {
-			warnOnErr(asyncEmitter.Close(), log)
-		}
+		warnOnErr(listener.Close(), log)
```

Comment:
Same as above, I'd add a comment with explanation here and everywhere below where applicable.

Reply:
Reduced PR scope.